src/dune_rules/obj_dir.ml

```diff
@@ -102,26 +106,43 @@ module Local = struct
     ; obj_dir : Path.Build.t
     ; native_dir : Path.Build.t
     ; byte_dir : Path.Build.t
+    ; melange_dir : Path.Build.t
```

Comment:
I'm not sure a single directory would be enough in the case where `private_modules` are used.

It would be worth it to add a test that exercises this feature in melange:

```
(library
 (private_modules bar)
 (name foo))
```

Reply:
Good to know. As next step, I am working on bringing over the tests from #1, but most probably will do so in a separate PR to not increase the size of this one.

I will include private modules tests in the melange suite 👍

Reply:
ftr: https://github.com/jchavarri/dune/pull/5

You were right, private_modules are broken. Looking at the code, I am not sure I understand why though. My initial question would be if this is something that would be fixed when installation support is added for melange libs? In which case, might be worth focusing on that? (rather than adding a fix for private modules in uninstalled libs)

Reply:
Installation support would be a separate thing. The reason why it's broken is roughly as follows: private modules must have their cmi's kept in a special directory. This is so that we're able to hide them from other libraries. In the current implementation, you only have one directory for melange cmi's, so you're unable to enforce the separation.